### ISSUE TITLE
com.hazelcast.spi.impl.AbstractDistributedObjectTest

### ISSUE THREAD

**Issue by Patras3** [hazelcast/hazelcast#22478]
_master_ (commit eef85cf6c332fe4f63455b8f06f51138f820b762)
Failed on RedHatJDK11: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-RedHatJDK11/53/testReport/com.hazelcast.spi.impl/AbstractDistributedObjectTest/

Below tests failed:
* testClusterVersion_isEqualTo_currentVersion
* testClusterVersion_isLessOrEqual_currentVersion
* testClusterVersion_isLessThan_nextMinorVersion
* testClusterVersion_isUnknownLessOrEqual_currentVersion
* testClusterVersion_isUnknownOrLessThan_nextMinorVersion

<details><summary>Stacktrace for testClusterVersion_isEqualTo_currentVersion as an example (issue seems be the same):</summary>

```
java.lang.AssertionError
	at org.junit.Assert.fail(Assert.java:87)
	at org.junit.Assert.assertTrue(Assert.java:42)
	at org.junit.Assert.assertTrue(Assert.java:53)
	at com.hazelcast.spi.impl.AbstractDistributedObjectTest.testClusterVersion_isEqualTo_currentVersion(AbstractDistributedObjectTest.java:57)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:829)

```
</details>

Similar issue we had for previous version: https://github.com/hazelcast/hazelcast/issues/20787

**Comment by ldziedziul** [comment]
Probably fixed with https://github.com/hazelcast/hazelcast/pull/22461. Could you re-check?

**Comment by krzysztofslusarski** [comment]
Fixed by #22461
